beacon-chain/rpc/prysm/v1alpha1/validator/proposer_attestations.go

```diff
@@ -271,16 +277,29 @@ func (a proposerAtts) sort() (proposerAtts, error) {
 	return a.sortBySlotAndCommittee()
 }
 
-func (a proposerAtts) sortOnChainAggregates() (proposerAtts, error) {
+func (a proposerAtts) sortOnChainAggregates(ctx context.Context, st state.ReadOnlyBeaconState) (proposerAtts, error) {
 	if len(a) < 2 {
 		return a, nil
 	}
 
-	// Sort by slot first, then by bit count.
+	totalBalance, err := helpers.TotalActiveBalance(st)
+	if err != nil {
+		return nil, err
+	}
+
+	// Sort attestation by proposer reward numerator
 	slices.SortFunc(a, func(a, b ethpb.Att) int {
```

Comment:
I am not sure how `SortFunc` works under the hood, but in case it's not caching the result of the sorting func, it will call `GetProposerRewardNumerator` multiple times for the same attestation. To optimize you can calculate the numerator once for all atts in advance and store in a `map[ethpb.Att]unit64`